mlflow/lightgbm.py

```diff
@@ -379,6 +379,11 @@ def log_feature_importance_plot(features, importance, importance_type):
         else:
             kwargs["callbacks"] = [callback]
 
+        # We set free_raw_data to false on the Dataset object
+        # so that we can access the original data later.
+        train_data = args[1] if len(args) > 1 else kwargs["train_data"]
+        train_data.free_raw_data = False
+
```

Comment:
Can we create a slice of `train_data.data` and use it after training instead of setting `free_raw_data` to `False`?

```python
sample = train_data.data[:5]

...

model = original(*args, **kwargs)  # == lgb.train

# do something using `sample`
```

Reply:
@harupy I think we wanted to run the original train function before attempting to slice the data in case of invalid input, i.e. if they put None, 2, etc as the data. The train function would perform its own validation on it and it would fail with a useful error message for the user.

However, if we just wrap this in a try-catch I think it should work nicely! And it also would save a lot of memory